java-snapshot-testing-core/src/main/java/au/com/origin/snapshots/ReflectionUtils.java

```diff
@@ -0,0 +1,48 @@
+package au.com.origin.snapshots;
+
+import lombok.experimental.UtilityClass;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+@UtilityClass
+public class ReflectionUtils {
```

Comment:
Should we move this util to a separate package, and add `@Internal` annotation?

Reply:
Yes - I think so.